source/common/common/logger.h

```diff
@@ -400,6 +401,43 @@ template <Id id> class Loggable {
     }                                                                                              \
   } while (0)
 
+#define ENVOY_LOG_FIRST_N(LEVEL, N, ...)                                                           \
+  do {                                                                                             \
+    static auto* countdown = new std::atomic<uint64_t>();                                          \
+    if (countdown->fetch_add(1) < N) {                                                             \
+      ENVOY_LOG(LEVEL, ##__VA_ARGS__);                                                             \
+    }                                                                                              \
+  } while (0)
+
+#define ENVOY_LOG_ONCE(LEVEL, ...)                                                                 \
+  do {                                                                                             \
+    ENVOY_LOG_FIRST_N(LEVEL, 1, ##__VA_ARGS__);                                                    \
+  } while (0)
+
+#define ENVOY_LOG_EVERY_NTH(LEVEL, N, ...)                                                         \
```

Comment:
As per earlier discussion, every Pow2 could be good as well.

Reply:
Oh yes good catch, my intent was actually to add that. Will do.

Reply:
Done, added `ENVOY_LOG_WITH_BACKOFF` in 371fa5dffe018d2a594008a07d242071e6e1424d + tests

Reply:
/cc @ggreenway ^^